src/amo/components/CollectionAddAddon/index.js

```diff
@@ -0,0 +1,207 @@
+  };
+
+  componentWillReceiveProps(props: InternalProps) {
+    const { hasAddonBeenAdded: hasAddonBeenAddedNew } = props;
+    const { hasAddonBeenAdded } = this.props;
+    if (hasAddonBeenAdded !== hasAddonBeenAddedNew) {
+      this.props.setUIState({
+        addonAddedStatus: props.hasAddonBeenAdded
+          ? ADDON_ADDED_STATUS_SUCCESS
+          : ADDON_ADDED_STATUS_NONE,
+      });
+    }
+
+    if (hasAddonBeenAddedNew && hasAddonBeenAddedNew !== hasAddonBeenAdded) {
+      this.timeout = this.props.setTimeout(
+        this.resetMessageStatus,
+        MESSAGE_RESET_TIME,
+      );
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.timeout) {
+      this.props.clearTimeout(this.timeout);
+    }
+  }
+
+  onAddonSelected = (suggestion: SuggestionType) => {
+    const {
+      collection,
+      currentUsername,
+      dispatch,
+      errorHandler,
+      filters,
+    } = this.props;
+    const { addonId } = suggestion;
+
+    invariant(addonId, 'addonId cannot be empty');
+    invariant(
+      collection,
+      'A collection must be loaded before you can add an add-on to it',
+    );
+    invariant(
+      currentUsername,
+      'Cannot add to collection because you are not signed in',
+    );
+
+    dispatch(
+      addAddonToCollection({
+        addonId,
+        collectionId: collection.id,
+        editing: true,
+        errorHandlerId: errorHandler.id,
+        filters,
+        slug: collection.slug,
+        username: currentUsername,
+      }),
+    );
+    this.props.setUIState({ addonAddedStatus: ADDON_ADDED_STATUS_PENDING });
+  };
+
+  timeout: TimeoutID;
+
+  resetMessageStatus = () => {
+    this.props.setUIState({
+      addonAddedStatus: ADDON_ADDED_STATUS_NONE,
+    });
+  };
+
+  render() {
+    const { errorHandler, i18n, uiState } = this.props;
+
+    return (
+      <div className="CollectionAddAddon">
+        {errorHandler.hasError() ? (
+          errorHandler.renderError()
+        ) : (
+          <TransitionGroup className="NoticePlaceholder">
+            {uiState.addonAddedStatus === ADDON_ADDED_STATUS_SUCCESS && (
+              <CSSTransition
+                classNames="NoticePlaceholder-transition"
+                timeout={MESSAGE_FADEOUT_TIME}
+              >
+                <Notice type="success">
+                  {i18n.gettext('Added to collection')}
+                </Notice>
+              </CSSTransition>
+            )}
+          </TransitionGroup>
+        )}
+
+        <AutoSearchInput
+          inputName="collection-addon-query"
+          inputPlaceholder={i18n.gettext(
+            'Find an add-on to include in this collection',
+          )}
+          onSearch={() => {}}
+          onSuggestionSelected={this.onAddonSelected}
+          selectSuggestionText={i18n.gettext('Add to collection')}
+        />
+      </div>
+    );
+  }
+}
+
+export const extractId = (props: Props) => {
+  const { collection } = props;
+  return collection ? collection.slug : '';
```

Comment:
For practical purposes this might be fine but switching to `collection.id` would be more stable.